src/helpers/requests/fetch-request.js

```diff
@@ -0,0 +1,43 @@
+import MiniPromise from '../../ponyfills/minipromise.js';
+import { createSignatureHeader } from './create-signature-header.js';
+
+
+const URL_PATTERN = /^([a-z]+):\/\/([^/]+)(.*)/i;
+
+
+export default function fetchRequest(options) {
+  return MiniPromise.createPromise(function(resolve, reject) {
```

Comment:
Is this library to like... add in promises to a js environment that doesn't have them?

Reply:
Yeah, that is the purpose. It's probably unnecessary now that we have ceased supporting IE11.